packages/protocol/contracts/layer1/core/libs/LibBondInstruction.sol

```diff
@@ -9,72 +9,46 @@ import { LibMath } from "src/shared/libs/LibMath.sol";
 library LibBondInstruction {
     using LibMath for uint256;
 
-    /// @dev Calculates bond instruction based on proposal age and proof timing windows.
-    ///
-    /// Bond logic:
-    /// - If proof is on-time (proposalAge <= provingWindow): No bond transfer
-    /// - If proof is late but within extended window: LIVENESS bond (designatedProver pays
-    ///   actualProver)
-    /// - If proof is after extended window: PROVABILITY bond (proposer pays actualProver)
-    /// - If payer == payee: No bond transfer
-    ///
+    /// @notice Calculates all bond instructions for a sequential prove call.
+    /// @dev Bond instruction rules:
+    ///         - On-time (within provingWindow + sequential grace): No bond changes.
+    ///         - Late: Liveness bond transfer, even when the designated and actual provers are the
+    ///           same address (L2 processing handles slashing/reward splits).
     /// @param _proposalId The proposal ID.
     /// @param _proposalTimestamp The proposal timestamp.
     /// @param _priorFinalizedTimestamp The timestamp when the last proposal was finalized.
-    /// @param _proposer The proposer address.
     /// @param _designatedProver The designated prover address.
     /// @param _actualProver The actual prover address.
     /// @param _provingWindow The proving window in seconds.
-    /// @param _extendedProvingWindow The extended proving window in seconds.
     /// @return bondInstruction_ A bond transfer instruction, or a BondType.NONE instruction when
     ///         no transfer is required.
     function calculateBondInstruction(
         uint48 _proposalId,
         uint256 _proposalTimestamp,
         uint48 _priorFinalizedTimestamp,
         uint48 _maxProofSubmissionDelay,
-        address _proposer,
         address _designatedProver,
         address _actualProver,
-        uint48 _provingWindow,
-        uint48 _extendedProvingWindow
+        uint48 _provingWindow
     )
         internal
         view
         returns (LibBonds.BondInstruction memory bondInstruction_)
     {
         uint256 proofTimestamp = block.timestamp;
         uint256 proposalDeadline = uint256(_proposalTimestamp) + _provingWindow;
-        uint256 sequentialDeadline =
-            uint256(_priorFinalizedTimestamp) + _maxProofSubmissionDelay;
+        uint256 sequentialDeadline = uint256(_priorFinalizedTimestamp) + _maxProofSubmissionDelay;
```

Comment:
now this lib function is very simple, I suggest we move it to Inbox.sol, then we don't have to pass in the two immutables (_maxProofSubmissionDelay &  _provingWindow) into the funciton.